### PR TITLE
add deck model / pw generation / pw authentication

### PR DESCRIPTION
**Pull request by ianagne**
@PacoGuy @KaraAJC @nimylian

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -4,4 +4,12 @@ class User < ActiveRecord::Base
 
 	validates :name, presence: true
```
Comment (KaraAJC) [inline app/models/user.rb:5]: should this not also be unique? how are we making sure users are unique?

Comment (ianagne) [inline app/models/user.rb:5]: I would vote adding an email field that is required to be unique.

Comment (KaraAJC) [inline app/models/user.rb:5]: i agree... @PacoGuy , @nimylian what do you think??

Comment (KaraAJC) [inline app/models/user.rb:5]: Shiraz agrees too!

**Pull request merged**